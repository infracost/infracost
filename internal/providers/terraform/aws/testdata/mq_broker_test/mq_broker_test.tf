provider "aws" {
  region                      = "us-east-1"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  skip_region_validation      = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

resource "aws_security_group" "my_aws_security_group" {
}

resource "aws_mq_configuration" "my_aws_mq_configuration" {
  description    = "Example Configuration"
  name           = "example"
  engine_type    = "ActiveMQ"
  engine_version = "5.15.0"

  data = <<DATA
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<broker xmlns="http://activemq.apache.org/schema/core">
  <plugins>
    <forcePersistencyModeBrokerPlugin persistenceFlag="true"/>
    <statisticsBrokerPlugin/>
    <timeStampingBrokerPlugin ttlCeiling="86400000" zeroExpirationOverride="86400000"/>
  </plugins>
</broker>
DATA
}

resource "aws_mq_configuration" "my_aws_mq_configuration_rabbitmq" {
  description    = "Example Configuration"
  name           = "example"
  engine_type    = "RabbitMQ"
  engine_version = "3.8.11"

  data = <<DATA
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<broker xmlns="http://activemq.apache.org/schema/core">
  <plugins>
    <forcePersistencyModeBrokerPlugin persistenceFlag="true"/>
    <statisticsBrokerPlugin/>
    <timeStampingBrokerPlugin ttlCeiling="86400000" zeroExpirationOverride="86400000"/>
  </plugins>
</broker>
DATA
}

resource "aws_mq_broker" "my_aws_mq_broker_activemq_single_default" {
  broker_name = "example"

  configuration {
    id       = aws_mq_configuration.my_aws_mq_configuration.id
    revision = aws_mq_configuration.my_aws_mq_configuration.latest_revision
  }

  engine_type        = "ActiveMQ"
  engine_version     = "5.15.9"
  host_instance_type = "mq.t2.micro"
  security_groups    = [aws_security_group.my_aws_security_group.id]
  deployment_mode    = "SINGLE_INSTANCE"

  user {
    username = "ExampleUser"
    password = "MindTheGappp"
  }
}

resource "aws_mq_broker" "my_aws_mq_broker_activemq_single_ebs" {
  broker_name = "example"

  configuration {
    id       = aws_mq_configuration.my_aws_mq_configuration.id
    revision = aws_mq_configuration.my_aws_mq_configuration.latest_revision
  }

  engine_type        = "ActiveMQ"
  engine_version     = "5.15.9"
  host_instance_type = "mq.t2.micro"
  security_groups    = [aws_security_group.my_aws_security_group.id]
  deployment_mode    = "SINGLE_INSTANCE"
  storage_type       = "ebs"

  user {
    username = "ExampleUser"
    password = "MindTheGappp"
  }
}

resource "aws_mq_broker" "my_aws_mq_broker_activemq_standby_efs" {
  broker_name = "example"

  configuration {
    id       = aws_mq_configuration.my_aws_mq_configuration.id
    revision = aws_mq_configuration.my_aws_mq_configuration.latest_revision
  }

  engine_type        = "ActiveMQ"
  engine_version     = "5.15.9"
  host_instance_type = "mq.m5.large"
  storage_type       = "efs"
  security_groups    = [aws_security_group.my_aws_security_group.id]
  deployment_mode    = "ACTIVE_STANDBY_MULTI_AZ"

  user {
    username = "ExampleUser"
    password = "MindTheGappp"
  }
}

resource "aws_mq_broker" "my_aws_mq_broker_rabbitmq_single" {
  broker_name = "example"

  configuration {
    id       = aws_mq_configuration.my_aws_mq_configuration.id
    revision = aws_mq_configuration.my_aws_mq_configuration.latest_revision
  }

  engine_type        = "RabbitMQ"
  engine_version     = "5.15.9"
  host_instance_type = "mq.m5.xlarge"
  security_groups    = [aws_security_group.my_aws_security_group.id]
  deployment_mode    = "SINGLE_INSTANCE"

  user {
    username = "ExampleUser"
    password = "MindTheGappp"
  }
}

resource "aws_mq_broker" "my_aws_mq_broker_rabbitmq_cluster" {
  broker_name = "example"

  configuration {
    id       = aws_mq_configuration.my_aws_mq_configuration_rabbitmq.id
    revision = aws_mq_configuration.my_aws_mq_configuration_rabbitmq.latest_revision
  }

  engine_type        = "RabbitMQ"
  engine_version     = "3.8.11"
  host_instance_type = "mq.m5.xlarge"
  security_groups    = [aws_security_group.my_aws_security_group.id]
  deployment_mode    = "cluster_multi_az"

  user {
    username = "ExampleUser"
    password = "MindTheGappp"
  }
}
