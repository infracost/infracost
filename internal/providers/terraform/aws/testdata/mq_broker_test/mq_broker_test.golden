
 Name                                                                  Monthly Qty  Unit            Monthly Cost 
                                                                                                                 
 aws_mq_broker.my_aws_mq_broker_activemq_single_default                                                          
 ├─ Instance usage (ActiveMQ, mq.t2.micro, single_instance)                    730  hours                 $21.90 
 └─ Storage (ActiveMQ, EFS)                                          Monthly cost depends on usage: $0.30 per GB 
                                                                                                                 
 aws_mq_broker.my_aws_mq_broker_activemq_single_ebs                                                              
 ├─ Instance usage (ActiveMQ, mq.t2.micro, single_instance)                    730  hours                 $21.90 
 └─ Storage (ActiveMQ, EBS)                                          Monthly cost depends on usage: $0.10 per GB 
                                                                                                                 
 aws_mq_broker.my_aws_mq_broker_activemq_standby_efs                                                             
 ├─ Instance usage (ActiveMQ, mq.m5.large, active_standby_multi_az)            730  hours                $420.48 
 └─ Storage (ActiveMQ, EFS)                                          Monthly cost depends on usage: $0.30 per GB 
                                                                                                                 
 aws_mq_broker.my_aws_mq_broker_rabbitmq_cluster                                                                 
 ├─ Instance usage (RabbitMQ, mq.m5.xlarge, cluster_multi_az)                  730  hours              $1,261.44 
 └─ Storage (RabbitMQ, EBS)                                                     30  GB                     $3.00 
                                                                                                                 
 aws_mq_broker.my_aws_mq_broker_rabbitmq_single                                                                  
 ├─ Instance usage (RabbitMQ, mq.m5.xlarge, single_instance)                   730  hours                $420.48 
 └─ Storage (RabbitMQ, EBS)                                          Monthly cost depends on usage: $0.10 per GB 
                                                                                                                 
 OVERALL TOTAL                                                                                         $2,149.20 
──────────────────────────────────
8 cloud resources were detected:
∙ 5 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 3 were free:
  ∙ 2 x aws_mq_configuration
  ∙ 1 x aws_security_group